File: 1784. check-if-binary-string-has-at-most-one-segment-of-ones.py
class Solution:
    def checkOnesSegment(self, s: str) -> bool:
        count = 0
        i = 0
        n = len(s)

        while i < n:
            if s[i] == '1':
                while i < n and s[i] == '1':
                    i += 1

                if count > 0:
                    return False

                count += 1
            else:
                i += 1

        return True

if __name__ == "__main__":
    sol = Solution()
    
    # Example 1:
    s = "1001"
    output = sol.checkOnesSegment(s)
    print(f"Example 1: Input: s = {s}")
    print(f"Output: {output}")
    print(f"Expected: False")
    print()
    
    # Example 2:
    s = "110"
    output = sol.checkOnesSegment(s)
    print(f"Example 2: Input: s = {s}")
    print(f"Output: {output}")
    print(f"Expected: True")

File: 1784. check-if-binary-string-has-at-most-one-segment-of-ones.java
class Solution {
    public boolean checkOnesSegment(String s) {
        int count = 0;
        int n = s.length();
        for (int i = 0; i < n; i++) {
            
            if (s.charAt(i) == '1') {
                while (i < n && s.charAt(i) == '1') {
                    i++;
                }
                if (count > 0)
                    return false;
                count++;
            }
        }

        return true;
    }
    
    public static void main(String[] args) {
        Solution sol = new Solution();
        
        // Example 1:
        String s1 = "1001";
        boolean output1 = sol.checkOnesSegment(s1);
        System.out.println("Example 1: Input: s = " + s1);
        System.out.println("Output: " + output1);
        System.out.println("Expected: false");
        System.out.println();
        
        // Example 2:
        String s2 = "110";
        boolean output2 = sol.checkOnesSegment(s2);
        System.out.println("Example 2: Input: s = " + s2);
        System.out.println("Output: " + output2);
        System.out.println("Expected: true");
    }
}